service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/IpPortManager.java

```diff
@@ -50,6 +50,12 @@
 
   private boolean autoDiscoveryInited = false;
 
+  private int maxRetryTimes;
+
+  public int getMaxRetryTimes() {
+    return maxRetryTimes;
```

Comment:
Maybe maxRetryTimes should be bigger than 2. Return Max(maxRestryTime, 2)？

Reply:
no need ,already retry twice